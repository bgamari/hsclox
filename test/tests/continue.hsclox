var a = 0;
while (a < 10) {
    a = a + 1;
    if (a > 5) {
        continue;
    }
    print a;
}

{
    var b = 5;
    while (b != 0) {
        b = b - 1;
        if (b == 3) {
            print "printing three";
            continue;
        }
    }
}

{
    var b = 5;
    while (b != 0) {
        b = b - 1;
        if (b == 3) {
            print "3!";
            continue;
        }

        if (b == 4) {
            print "4!";
            continue;
        }
    }
}

{
    var b = 5;
    var g = "scopes ok";
    while (b != 0) {
        var x = 6;
        var y = 0;
        b = b - 1;
        if (b == 0) {
            print "zero";
            continue;
        }
    }
    print g;
}

{
    for (var b = 0; b != 5; b = b + 1) {
        if (b == 0 or b == 1 or b == 2) {
            print b;
            continue;
        }
    }
}
