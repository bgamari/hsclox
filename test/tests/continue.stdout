== code ==
0000    1 OP_CONSTANT         1 '0
0002    | OP_DEFINE_GLOBAL    0 'a
0004    2 OP_GET_GLOBAL       0 'a
0006    | OP_CONSTANT         2 '10
0008    | OP_LESS
0009    | OP_JUMP_IF_FALSE    9 -> 43
0012    | OP_POP
0013    3 OP_GET_GLOBAL       0 'a
0015    | OP_CONSTANT         3 '1
0017    | OP_ADD
0018    | OP_SET_GLOBAL       0 'a
0020    | OP_POP
0021    4 OP_GET_GLOBAL       0 'a
0023    | OP_CONSTANT         4 '5
0025    | OP_GREATER
0026    | OP_JUMP_IF_FALSE   26 -> 36
0029    | OP_POP
0030    5 OP_LOOP            30 -> 4
0033    6 OP_JUMP            33 -> 37
0036    | OP_POP
0037    7 OP_GET_GLOBAL       0 'a
0039    | OP_PRINT
0040    8 OP_LOOP            40 -> 4
0043    | OP_POP
0044   11 OP_CONSTANT         5 '5
0046   12 OP_GET_LOCAL     04
0048    | OP_CONSTANT         6 '0
0050    | OP_EQUAL
0051    | OP_NOT
0052    | OP_JUMP_IF_FALSE   52 -> 86
0055    | OP_POP
0056   13 OP_GET_LOCAL     04
0058    | OP_CONSTANT         7 '1
0060    | OP_SUBTRACT
0061    | OP_SET_LOCAL     04
0063    | OP_POP
0064   14 OP_GET_LOCAL     04
0066    | OP_CONSTANT         8 '3
0068    | OP_EQUAL
0069    | OP_JUMP_IF_FALSE   69 -> 82
0072    | OP_POP
0073   15 OP_CONSTANT         9 'printing three
0075    | OP_PRINT
0076   16 OP_LOOP            76 -> 46
0079   17 OP_JUMP            79 -> 83
0082    | OP_POP
0083   18 OP_LOOP            83 -> 46
0086    | OP_POP
0087   19 OP_POP
0088   22 OP_CONSTANT        10 '5
0090   23 OP_GET_LOCAL     04
0092    | OP_CONSTANT        11 '0
0094    | OP_EQUAL
0095    | OP_NOT
0096    | OP_JUMP_IF_FALSE   96 -> 149
0099    | OP_POP
0100   24 OP_GET_LOCAL     04
0102    | OP_CONSTANT        12 '1
0104    | OP_SUBTRACT
0105    | OP_SET_LOCAL     04
0107    | OP_POP
0108   25 OP_GET_LOCAL     04
0110    | OP_CONSTANT        13 '3
0112    | OP_EQUAL
0113    | OP_JUMP_IF_FALSE  113 -> 126
0116    | OP_POP
0117   26 OP_CONSTANT        14 '3!
0119    | OP_PRINT
0120   27 OP_LOOP           120 -> 90
0123   28 OP_JUMP           123 -> 127
0126    | OP_POP
0127   30 OP_GET_LOCAL     04
0129    | OP_CONSTANT        15 '4
0131    | OP_EQUAL
0132    | OP_JUMP_IF_FALSE  132 -> 145
0135    | OP_POP
0136   31 OP_CONSTANT        16 '4!
0138    | OP_PRINT
0139   32 OP_LOOP           139 -> 90
0142   33 OP_JUMP           142 -> 146
0145    | OP_POP
0146   34 OP_LOOP           146 -> 90
0149    | OP_POP
0150   35 OP_POP
0151   38 OP_CONSTANT        17 '5
0153   39 OP_CONSTANT        18 'scopes ok
0155   40 OP_GET_LOCAL     04
0157    | OP_CONSTANT        19 '0
0159    | OP_EQUAL
0160    | OP_NOT
0161    | OP_JUMP_IF_FALSE  161 -> 201
0164    | OP_POP
0165   41 OP_CONSTANT        20 '6
0167   42 OP_CONSTANT        21 '0
0169   43 OP_GET_LOCAL     04
0171    | OP_CONSTANT        22 '1
0173    | OP_SUBTRACT
0174    | OP_SET_LOCAL     04
0176    | OP_POP
0177   44 OP_GET_LOCAL     04
0179    | OP_CONSTANT        23 '0
0181    | OP_EQUAL
0182    | OP_JUMP_IF_FALSE  182 -> 197
0185    | OP_POP
0186   45 OP_CONSTANT        24 'zero
0188    | OP_PRINT
0189   46 OP_POPN             2 '10
0191    | OP_LOOP           191 -> 155
0194   47 OP_JUMP           194 -> 198
0197    | OP_POP
0198   48 OP_LOOP           198 -> 155
0201    | OP_POP
0202   49 OP_GET_LOCAL     14
0204    | OP_PRINT
0205   50 OP_POPN             2 '10
0207   53 OP_CONSTANT        25 '0
0209    | OP_GET_LOCAL     04
0211    | OP_CONSTANT        26 '5
0213    | OP_EQUAL
0214    | OP_NOT
0215    | OP_JUMP_IF_FALSE  215 -> 279
0218    | OP_POP
0219    | OP_JUMP           219 -> 233
0222    | OP_GET_LOCAL     04
0224    | OP_CONSTANT        27 '1
0226    | OP_ADD
0227    | OP_SET_LOCAL     04
0229    | OP_POP
0230    | OP_LOOP           230 -> 209
0233   54 OP_GET_LOCAL     04
0235    | OP_CONSTANT        28 '0
0237    | OP_EQUAL
0238    | OP_JUMP_IF_FALSE  238 -> 244
0241    | OP_JUMP           241 -> 262
0244    | OP_POP
0245    | OP_GET_LOCAL     04
0247    | OP_CONSTANT        29 '1
0249    | OP_EQUAL
0250    | OP_JUMP_IF_FALSE  250 -> 256
0253    | OP_JUMP           253 -> 262
0256    | OP_POP
0257    | OP_GET_LOCAL     04
0259    | OP_CONSTANT        30 '2
0261    | OP_EQUAL
0262    | OP_JUMP_IF_FALSE  262 -> 275
0265    | OP_POP
0266   55 OP_GET_LOCAL     04
0268    | OP_PRINT
0269   56 OP_LOOP           269 -> 222
0272   57 OP_JUMP           272 -> 276
0275    | OP_POP
0276   58 OP_LOOP           276 -> 222
0279    | OP_POP
0280    | OP_POP
0281   60 OP_RETURN
1
2
3
4
5
printing three
4!
3!
zero
scopes ok
0
1
2
